api/chat_main_service.go

```diff
@@ -235,3 +235,45 @@ func GetAiAnswerOpenApi(msgs []openai.ChatCompletionMessage) (ChatCompletionResp
 	}
 	return aiAnswer, nil
 }
+
+func (s *ChatService) getAskMessages(chatSession sqlc_queries.ChatSession, chatUuid string, regenerate bool) ([]openai.ChatCompletionMessage, error) {
+	ctx := context.Background()
+	chatSessionUuid := chatSession.Uuid
+
+	lastN := chatSession.MaxLength
+	if chatSession.MaxLength == 0 {
+		lastN = 10
+	}
+
+	chat_prompts, err := s.q.GetChatPromptsBySessionUUID(ctx, chatSessionUuid)
+
+	if err != nil {
+		return nil, fmt.Errorf("fail to get prompt: %w", err)
+	}
+
+	var chat_massages []sqlc_queries.ChatMessage
+	if regenerate {
+		chat_massages, err = s.q.GetLastNChatMessages(ctx,
+			sqlc_queries.GetLastNChatMessagesParams{
+				Uuid:  chatUuid,
+				Limit: lastN,
+			})
+
+	} else {
+		chat_massages, err = s.q.GetLatestMessagesBySessionUUID(ctx,
+			sqlc_queries.GetLatestMessagesBySessionUUIDParams{ChatSessionUuid: chatSession.Uuid, Limit: lastN})
+	}
+
+	if err != nil {
+		return nil, fmt.Errorf("fail to get messages: %w", err)
+	}
+	chat_prompt_msgs := lo.Map(chat_prompts, func(m sqlc_queries.ChatPrompt, _ int) openai.ChatCompletionMessage {
+		return openai.ChatCompletionMessage{Role: m.Role, Content: m.Content}
+	})
+	chat_message_msgs := lo.Map(chat_massages, func(m sqlc_queries.ChatMessage, _ int) openai.ChatCompletionMessage {
+		return openai.ChatCompletionMessage{Role: m.Role, Content: m.Content}
+	})
+	msgs := append(chat_prompt_msgs, chat_message_msgs...)
+	return msgs, nil
+}
+
```

Comment:
:robot: OpenAI

This patch adds a new function `getAskMessages` to the `ChatService` struct. The function retrieves chat prompts and messages based on session UUID and maximum length.

The implementation looks good to me. However, I have a few suggestions:

1. It would be better to use a context with a timeout instead of a background context to avoid blocking indefinitely in case of a database query failure.
2. The error message returned by the function should include more details about the error that occurred, such as the name of the function that failed and the input parameters.

Here's an updated version of the patch with these changes:

```diff
@@ -235,3 +235,45 @@ func GetAiAnswerOpenApi(msgs []openai.ChatCompletionMessage) (ChatCompletionResp
 	}
 	return aiAnswer, nil
 }
+
+func (s *ChatService) getAskMessages(ctx context.Context, chatSession sqlc_queries.ChatSession, chatUuid string, regenerate bool) ([]openai.ChatCompletionMessage, error) {
+	ctx, cancel := context.WithTimeout(ctx, time.Second*10)
+	defer cancel()
+
+	chatSessionUuid := chatSession.Uuid
+
+	lastN := chatSession.MaxLength
+	if chatSession.MaxLength == 0 {
+		lastN = 10
+	}
+
+	chat_prompts, err := s.q.GetChatPromptsBySessionUUID(ctx, chatSessionUuid)
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to get chat prompts: %w", err)
+	}
+
+	var chat_massages []sqlc_queries.ChatMessage
+	if regenerate {
+		chat_massages, err = s.q.GetLastNChatMessages(ctx,
+			sqlc_queries.GetLastNChatMessagesParams{
+				Uuid:  chatUuid,
+				Limit: lastN,
+			})
+
+	} else {
+		chat_massages, err = s.q.GetLatestMessagesBySessionUUID(ctx,
+			sqlc_queries.GetLatestMessagesBySessionUUIDParams{ChatSessionUuid: chatSession.Uuid, Limit: lastN})
+	}
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to get chat messages: %w", err)
+	}
+	chat_prompt_msgs := lo.Map(chat_prompts, func(m sqlc_queries.ChatPrompt, _ int) openai.ChatCompletionMessage {
+		return openai.ChatCompletionMessage{Role: m.Role, Content: m.Content}
+	})
+	chat_message_msgs := lo.Map(chat_massages, func(m sqlc_queries.ChatMessage, _ int) openai.ChatCompletionMessage {
+		return openai.ChatCompletionMessage{Role: m.Role, Content: m.Content}
+	})
+	msgs := append(chat_prompt_msgs, chat_message_msgs...)
+	return msgs, nil
+}
+
``` 

Please let me know if you have any questions or concerns.

<!-- This is an auto-generated comment by OpenAI -->